packages/@sfx/search-driver-plugin/src/search-driver-plugin.ts

```diff
@@ -100,8 +119,31 @@ export default class SearchDriverPlugin implements Plugin {
    *
    * @param query the query to send.
    */
-  sendSearchApiRequest(query: string): Promise<Results> {
+  sendSearchApiRequest(query: string): Promise<SearchResponseSection<P>> {
     const fullQuery = { ...this.defaultSearchConfig, query };
-    return this.core.search.search(fullQuery);
+    return this.core.search.search(fullQuery)
+      .then(this.searchCallback);
+  }
+
+  /**
+   * Extracts query and products from the given response.
+   * Calls `this.transformProduct` on each product found in the response.
+   * Filters out any products that map to a falsy value.
+   *
+   * @param response An object containing the original query and product records.
+   * @returns An object containing the query and an array of valid simplified products.
+   */
+  searchCallback(response: Results): SearchResponseSection<P> {
+    const { records } = response;
+    const mappedRecords = records.map(this.transformProduct).filter(Boolean);
+
+    return {
+      originalResponse: response,
+      products: mappedRecords,
+    };
   }
 }
+
+export interface SearchDriverOptions {
```

Comment:
Like with the SAYT driver, this needs a generic:

```suggestion
export interface SearchDriverOptions<P> {
```

Add docs:

```ts
/**
 * The configuration options for [[SearchDriver]].
 *
 * @typeparam P The product type.
 */
```